### PR TITLE
Combine wrapper and final layer

### PR DESCRIPTION
**Pull request by Dimo99**

### CODE REVIEW

File: beacon-light-client/plonky2/circuits_executables/bin/wrapper.rs [hunk 1]
```diff
@@ -51,81 +48,25 @@ async fn async_main() -> Result<()> {
...
-
-        con.publish("gnark_proofs_channel", "start").await?;
-    }
+    wrap_final_layer_in_poseidon_bn_128(
+        con,
+        compile_circuit,
+        final_layer_circuit,
+        final_layer_proof,
+    )
+    .await?;
```
Comment (Xearty) [inline beacon-light-client/plonky2/circuits_executables/bin/wrapper.rs:69]: This function does too much. I would extract it to a different file and make it only build the circuit

File: beacon-light-client/plonky2/circuits_executables/src/wrap_final_layer_in_poseidon_bn128.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        &proof_target.clone(),
+        &verifier_target,
+        &final_layer_circuit.common,
+    );
+    builder.register_public_inputs(&proof_target.public_inputs);
+    let mut pw = PartialWitness::new();
+    pw.set_proof_with_pis_target(&proof_target, &final_layer_proof);
+    let circuit_data = builder.build::<PoseidonBN128GoldilocksConfig>();
+    let proof = circuit_data.prove(pw)?;
+    let proof = serde_json::to_string(&proof)?;
```
Comment (Xearty) [inline beacon-light-client/plonky2/circuits_executables/src/wrap_final_layer_in_poseidon_bn128.rs:48]: Try making this a single line or rename the original proof

Comment (Xearty) [reply beacon-light-client/plonky2/circuits_executables/src/wrap_final_layer_in_poseidon_bn128.rs:48]: Let's switch to using the `CommandLineOptionsBuilder` for `final_layer` in this PR

**Comment by Xearty** [comment]
Let's switch to using the CommandLineOptionsBuilder for final_layer in this PR

File: beacon-light-client/plonky2/circuits_executables/src/wrap_final_layer_in_poseidon_bn128.rs (hunk 2 above)
Comment (Xearty) [reply beacon-light-client/plonky2/circuits_executables/src/wrap_final_layer_in_poseidon_bn128.rs:48]: I would use the name `serialized_proof` or `marshalled_proof`

**Review by Xearty — APPROVED** [review]

**Review by Xearty — APPROVED** [review]

**Review by EmilIvanichkovv — APPROVED** [review]

**Comment by Dimo99** [comment]
> Delete wrapper.rs

We agreed not to delete it because it is used for compiling the wrapper circuit and generating json files needed for the gnark server

**Pull request merged**